plugins/quote.py

```diff
@@ -146,6 +146,53 @@ def get_quote_by_chan(db, chan, num=False):
     return format_quote(data, num, count)
 
 
+@hook.command('findquote')
+def find_quote(db, text, notice):
+    """[#chan] [nick] <to match> OR [nick] <to match> - get the first quote matching <to match> by [nick] in [#chan]"""
+
+    errstr = "No quotes found"
+    find = re.match(r"(?:(#\S+)\s+)?(\S+)\s+(.*)$", text)
```

Comment:
Avoid recreating the regex on each call by declaring and compiling the regex as a module level variable.